actors/market/src/lib.rs

```diff
@@ -234,16 +235,34 @@ impl Actor {
             ));
         }
 
+        struct CheckedDeal {
```

Comment:
This is only needed to maintain the index of the deals that are determined to be valid. I dislike doing this, and would welcome a better suggestion.

One option is to include the `original_index` in some struct (instead of `check_passed`, but that feels just as bad.

Reply:
You don't need to copy the proposals. A parallel `vec<bool>` would suffice.

Reply:
That sounds good, thank you!